planning/autoware_obstacle_cruise_planner/src/node.cpp

```diff
@@ -1454,71 +1652,81 @@ std::optional<StopObstacle> ObstacleCruisePlannerNode::createStopObstacleForPred
-    const auto [obstacle_tangent_vel, obstacle_normal_vel] =
-      projectObstacleVelocityToTrajectory(traj_points, obstacle);
-    if (p.obstacle_velocity_threshold_from_stop_to_cruise <= obstacle_tangent_vel) {
+    if (!isOutsideStopObstacle(obstacle.classification.label)) {
       return std::nullopt;
     }
-  }
 
-  // NOTE: Dynamic obstacles for stop are crossing ego's trajectory with high speed,
-  //       and the collision between ego and obstacles are within the margin threshold.
-  const bool is_obstacle_crossing = isObstacleCrossing(traj_points, obstacle);
-  const bool has_high_speed = p.crossing_obstacle_velocity_threshold <
-                              std::hypot(obstacle.twist.linear.x, obstacle.twist.linear.y);
-  if (is_obstacle_crossing && has_high_speed) {
-    // Get highest confidence predicted path
-    const auto resampled_predicted_path = resampleHighestConfidencePredictedPath(
-      obstacle.predicted_paths, p.prediction_resampling_time_interval,
-      p.prediction_resampling_time_horizon);
-
-    std::vector<size_t> collision_index;
-    const auto collision_points = polygon_utils::getCollisionPoints(
-      traj_points, traj_polys, obstacle.stamp, resampled_predicted_path, obstacle.shape, now(),
-      is_driving_forward_, collision_index,
-      calcObstacleMaxLength(obstacle.shape) + p.decimate_trajectory_step_length +
-        std::hypot(
-          vehicle_info_.vehicle_length_m,
-          vehicle_info_.vehicle_width_m * 0.5 + max_lat_margin_for_stop));
-    if (collision_points.empty()) {
+    const auto time_to_traj = precise_lat_dist / std::max(1e-6, obstacle.normal_velocity);
+    if (time_to_traj > p.max_lat_time_margin_for_stop) {
       RCLCPP_INFO_EXPRESSION(
         get_logger(), enable_debug_info_,
-        "[Stop] Ignore inside obstacle (%s) since there is no collision point between the "
-        "predicted path "
-        "and the ego.",
-        object_id.c_str());
-      debug_data_ptr_->intentionally_ignored_obstacles.push_back(obstacle);
+        "[Stop] Ignore outside obstacle (%s) since it's far from trajectory.", object_id.c_str());
       return std::nullopt;
     }
 
-    const double collision_time_margin =
-      calcCollisionTimeMargin(odometry, collision_points, traj_points, is_driving_forward_);
-    if (p.collision_time_margin < collision_time_margin) {
-      RCLCPP_INFO_EXPRESSION(
-        get_logger(), enable_debug_info_,
-        "[Stop] Ignore inside obstacle (%s) since it will not collide with the ego.",
-        object_id.c_str());
-      debug_data_ptr_->intentionally_ignored_obstacles.push_back(obstacle);
+    // brkay54: For the pedestrians and bicycles, we need to check the collision point by thinking
+    // they will stop with a predefined deceleration rate to avoid unnecessary stops.
+    double resample_time_horizon = p.prediction_resampling_time_horizon;
+    if (obstacle.classification.label == ObjectClassification::PEDESTRIAN) {
+      resample_time_horizon =
```

Comment:
For the case to calculate the time to stop, sure tihs `velocty / -acceleration` is correct, however, I think we want to calculate resample_time_horizon as a argument value to `resampleHighestConfidencePredictedPaths()`.
`resampleHighestConfidencePredictedPaths()` assumes the object moves with constant velocity (or twist), but we are assuming the pedestrians are decelerate constantly.
Therefore should we change the formula as `velocty / (2.0 * -acceleration)`?